src/metabase/sync/sync_metadata/fields/our_metadata.clj

```diff
@@ -19,23 +19,28 @@
 (mu/defn- fields->parent-id->fields :- [:map-of common/ParentID [:set common/TableMetadataFieldWithID]]
   [fields :- [:maybe [:sequential i/FieldInstance]]]
   (->> (for [field fields]
-         {:parent-id                 (:parent_id field)
-          :id                        (:id field)
-          :name                      (:name field)
-          :database-type             (:database_type field)
-          :effective-type            (:effective_type field)
-          :coercion-strategy         (:coercion_strategy field)
-          :base-type                 (:base_type field)
-          :semantic-type             (:semantic_type field)
-          :pk?                       (isa? (:semantic_type field) :type/PK)
-          :field-comment             (:description field)
-          :json-unfolding            (:json_unfolding field)
-          :database-is-auto-increment (:database_is_auto_increment field)
-          :position                  (:position field)
-          :database-position         (:database_position field)
-          :database-partitioned      (:database_partitioned field)
-          :database-required         (:database_required field)
-          :visibility-type           (:visibility_type field)})
+         (merge
+          {:parent-id                  (:parent_id field)
+           :id                         (:id field)
+           :name                       (:name field)
+           :database-type              (:database_type field)
+           :effective-type             (:effective_type field)
+           :coercion-strategy          (:coercion_strategy field)
+           :base-type                  (:base_type field)
+           :semantic-type              (:semantic_type field)
+           :field-comment              (:description field)
+           :json-unfolding             (:json_unfolding field)
+           :database-is-auto-increment (:database_is_auto_increment field)
+           :position                   (:position field)
+           :database-position          (:database_position field)
+           :database-partitioned       (:database_partitioned field)
+           :database-required          (:database_required field)
+           :visibility-type            (:visibility_type field)}
+          (u/remove-nils
+           {:pk?                   (:database_is_pk field)
```

Comment:
the new definition of :database_is_pk will cause all fields to be updated on a first run to save the new value of :database_is_pk where pks are defined (nil -> true/false). 

There may be some other unintended consequence of this change. I did consider keeping the old semantic type expression (perhaps in an `(or)`). Feel free to reconsider this.